source/dub/recipe/selection.d

```diff
@@ -44,6 +44,18 @@ public struct SelectionsFile
         return this.content.match!((s) => s.fileVersion);
     }
 
+    /**
+     * Whether this dub.selections.json can be inherited by nested projects
+     * without local dub.selections.json
+     */
+    public bool inheritable () const @safe pure nothrow @nogc
+    {
+        return this.content.match!(
+            (const Selections!0 _) => false,
+            (const Selections!1 s) => s.inheritable,
+        );
+    }
+
```

Comment:
I don't know if we should expose this convenience function here. The intent of `SelectionFile` is just to wrap a specific `Selections!uint`. However, the implementation makes sense logically. IMO the question boils down to: "Do we ever find ourselves in a situation where we manipulate an invalid `Selections!uint`?". I would be tempted to say no, as trying to make a generic `Dependency` instead of the discriminated union it is now is what made the code a mess.